model/src/main/kotlin/Identifier.kt

```diff
@@ -124,4 +124,5 @@ data class Identifier(
     //       https://github.com/package-url/purl-spec/issues/33.
     fun toPurl() = "".takeIf { this == EMPTY }
         ?: "pkg:${type.toLowerCase()}/${namespace.percentEncode()}/${name.percentEncode()}@${version.percentEncode()}"
+            .replace("//", "/")
```

Comment:
How about using a string builder instead?

```kotlin
buildString {
  append("pkg:")
  append(type.toLowerCase())
  if (namespace.isNotBlank()) {
    append("/")
    append(namespace.percentEncode())
  }
  ...
}
```

More code, but also more expressive.

Reply:
Will do, though my ultimate goal is to switch to https://github.com/package-url/packageurl-java, but from a quick glance it seems that library needs some improvements.